docs/overview.md

```diff
@@ -0,0 +1,100 @@
+network will be used to gossip Parachain blocks and to gossip transactions. Collators will only
+gossip blocks to the Parachain network that have a high chance of being included in the relay
+chain. To prove that a block is probably going to be included, the Collator will send along side
+the notification the so-called candidate message. This candidate message is issued by a Parachain
+validator after approving a block. This proof of possible inclusion prevents spamming other collators
+of the network with useless blocks.
+The Collator joins the relay-chain network for two reasons. First, the Collator uses it to send the
+Parachain blocks to the Parachain validators. Secondly, the Collator participates as light/full-node
+of the relay chain to be informed of new relay-chain blocks. This information will be used for the
+consensus and the block production logic.
+
+## Block building
+
+Polkadot requires that a Parachain block is transmitted in a fixed format. These blocks sent by a
+Parachain to the Parachain validators are called proof-of-validity blocks (PovBlock). Such a
+PovBlock contains the header and the transactions of the Parachain as opaque blobs (`Vec<u8>`). They
+are opaque, because Polkadot can not and should not support all kinds of possible Parachain block
+formats. Besides the header and the transactions, it also contains the witness data and the outgoing
+messages.
+
+A Parachain validator needs to validate a given PoVBlock, but without requiring the full state of
+the Parachain. To still make it possible to validate the Parachain block, the PovBlock contains the
+witness data. The witness data is a proof that is collected while building the block. The proof will
+contain all trie nodes that are read during the block production. Cumulus uses the witness data to
+reconstruct a partial trie and uses this a storage when executing the block.
+
+The outgoing messages are also collected at block production. These are messages from the Parachain
+the block is built for to other Parachains or to the relay chain itself.
+
+## Runtime Upgrade
+
+Every Substrate blockchain supports runtime upgrades. Runtime upgrades enable a blockchain to update
+its state transition function without requiring any client update. Such a runtime upgrade is applied
+by a special transaction in a Substrate runtime. Polkadot and Cumulus provide support for these
+runtime upgrades, but updating a Parachain runtime is not as easy as updating a standalone
+blockchain runtime. In a standalone blockchain, the special transaction needs to be included in a
+block and the runtime is updated.
+
+A Parachain will follow the same paradigm, but the relay chain needs to be informed before
+the update. Cumulus will provide functionality to notify the relay chain about the runtime update. The
+update will not be enacted directly; instead it takes `X` blocks (a value that is configured by the relay
```

Comment:
This isn't exactly accurate - see https://github.com/paritytech/polkadot/issues/825 . It'll be more like "applied after the first included parablock executed in a relay chain context X blocks from now". But that is definitely a complicated sentence for docs, so it would make sense to find a way to simplify.